### PR TITLE
Set clang-format cpp style to modern C++

### PR DESCRIPTION
**Pull request by Meakk**
Let's bump C++ version to `Cpp11` (which means C++11 and above).  
It mainly impacts the closing bracket of nested template argument but I'll need it for a upcoming PR where I'm using raw strings.

### CODE REVIEW

**Comment by snoyer** [comment]
here's a spot where `Cpp03` formatting had to be disabled but should work with `Cpp11` https://github.com/f3d-app/f3d/blob/b37f45513094f3b28eb7c264ab056281be6b130f/library/src/image.cxx#L515-L519

**Comment by Meakk** [comment]
Indeed! Let me fix that.

**Comment by Meakk** [comment]
@mwestphal please review

File: library/src/options.cxx [hunk 1]
```diff
@@ -390,13 +390,13 @@ std::string options::getAsString(const std::string& name) const
 //----------------------------------------------------------------------------
 std::vector<int> options::getAsIntVector(const std::string& name) const
 {
-  return this->Internals->get<std::vector<int> >(name);
+  return this->Internals->get<std::vector<int>>(name);
```
Comment (mwestphal) [inline library/src/options.cxx:393]: this being uncovered is unexpected, is it not ?

Comment (mwestphal) [reply library/src/options.cxx:393]: or maybe it is.

**Review by mwestphal — APPROVED** [review]

**Pull request merged**